installers/AppVeyor.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE NoImplicitPrelude, OverloadedStrings, LambdaCase, RecordWildCards, DeriveDataTypeable, DeriveGeneric #-}
+
+module AppVeyor (downloadCardanoSL, AppVeyorError(..)) where
+
+import Universum hiding (get)
+import Data.Aeson
+import GitHub hiding (URL)
+import GitHub.Endpoints.Repos.Statuses
+import qualified GitHub as GH
+import qualified Data.ByteString.Lazy.Char8 as L8
+import qualified Data.Text as T
+import Network.URI
+import Data.Maybe (mapMaybe)
+import Network.Wreq
+import Data.Aeson.Lens
+import Lens.Micro
+import Turtle.Format (printf, (%), d, s, w, Format, makeFormat)
+import GHC.Generics
+
+-- | Gets CardanoSL.zip corresponding to the src json revision from AppVeyor CI
+downloadCardanoSL :: FilePath -> IO L8.ByteString
+downloadCardanoSL srcJson = do
+  src@CardanoSource{..} <- getCardanoRev srcJson
+  printf ("Fetching GitHub CI status for commit "%n%" of "%n%"/"%n%"\n") srcRev srcOwner srcRepo
+  buildURL <- appVeyorURL src
+  printf ("Build URL is "%s%"\n") (getUrl buildURL)
+  jobs <- appVeyorJobs buildURL
+  printf ("Build has "%d%" job(s), status "%ss%".\n") (length jobs)
+    (map (show . jobStatus) jobs)
+  arts <- appVeyorArtifacts $ map jobId jobs
+  printf ("Artifacts are: "%ss%"\n") (map fst arts)
+  case findAppVeyorArtifact "CardanoSL.zip" arts of
+    Just bs -> printf "Done\n" >> pure bs
+    Nothing -> throwM (MissingArtifactsError $ getUrl buildURL)
+
+n :: Format r (Name a -> r)
+n = makeFormat untagName
+
+ss :: Format r ([Text] -> r)
+ss = makeFormat (T.intercalate ",")
```

Comment:
I think it might be worth starting to collect these little utility functions together in some kind of a `Misc` file -- together with or as a replacement of https://github.com/input-output-hk/daedalus/blob/develop/installers/Types.hs#L82

Reply:
`n` is specific to `GitHub.Name` type and `ss` hardcodes the separator string, so no.